### DISCUSSION TITLE
Focus lost when using AlpineJS $id after the Livewire request is sent

### DISCUSSION THREAD

**Discussion by johanvanhelden** [livewire/livewire#4790] [1. Help]
Hi all! Not sure if this is a bug, or me not understanding something.

I was watching Caleb's Laracon talk and he was showing of how to use Alpine to generate ids for elements.

So I just finished implementing it, but after I type some data into an input element, and the Livewire request is sent to the server, the input looses its focus.

**Demo:**
![LivewireAlpineDemo](https://user-images.githubusercontent.com/19389981/157751947-bbf80441-2eb7-4e46-a9d1-15b1fa32fa1c.gif)

The HTML for the input element looks like this:
```html
<div x-data="" x-id="['input']">
    <label class="block | text-sm font-medium text-gray-700 mb-2" x-bind:for="$id('input')" for="input-2">Name</label>
    <div>
        <input 
            x-bind:id="$id('input')" 
            type="text" 
            class="w-full block | shadow-sm rounded border-gray-300 sm:text-sm | focus:ring-blue-500 focus:border-blue-500" 
            wire:model="state.name" 
            autocomplete="name" 
            autofocus="autofocus" 
            id="input-2"
        >
    </div>
</div>
```

If I revert the `$id()` changes, everything works like it should, without anything loosing focus.

**Comment by joshhanley** [comment]
@johanvanhelden probably a bug, but I'm not sure if that's a Livewire issue or not. When you inspect the element in devtools, and see what ID it has generated (before a Livewire request), then enter something into your input and inspect the element again, does the ID change?

**Reply by johanvanhelden** [reply]
Thanks for your help @joshhanley ! Unfortunately the ID stays the same. Before the reload it is `input-2` and after losing the focus it remains `input-2`. So that one does not change. I can actually not see any changes in the source. No part of the input group's HTML seems to change.

**Comment by davisngl** [comment]
Having absolutely the same issue and same way how I got to know about `$id()` feature lol.

For now, it seems that it's only issue with $id(... as I narrowed down the suspicious Blade components that might cause it, but it only got fixed after removing id binding on the input.

And the actual id doesn't ever change after render.

Maybe you have fixed it, @johanvanhelden?

**Comment by davisngl** [comment]
Well, the only thing, that worked, was using `.lazy` modifier for `wire:model` when `$id()` function is used from Alpine.

Any comments, @calebporzio ? Ready to provide a reproducible sample, if necessary.

**Comment by mmerlijn** [comment]
I had the same issue, and used `wire:key` in the input as work around.

**Comment by joshhanley** [comment]
I think this issue should be raised on the Alpine repo as the code for it is managed there and not here. I don't know how the `$id` function works under the hood, so can't be of much help, but someone on the Alpine repo might have a better idea of what might be going on 🙂

**Reply by davisngl** [reply]
That was already brought up in the AlpineJS repository, but to no avail. https://github.com/alpinejs/alpine/discussions/2826

**Reply by joshhanley** [reply]
@davisngl hmm ok, I would have thought one of the Alpine guys would have had a look.

**Reply by johanvanhelden** [reply]
Just to be sure I've raised an issue in the AlpineJS/Alpine repo: https://github.com/alpinejs/alpine/issues/3097

**Comment by dakira** [comment]
Oh wow. This happens to me, too. I really love Alpines `$id()` but I guess I'll have to stop using it now, unless I want to at wire:key to every single input element in my apps.
